src/librustc_codegen_ssa/back/link.rs

```diff
@@ -1356,8 +1356,18 @@ fn add_link_script(cmd: &mut dyn Linker, sess: &Session, tmpdir: &Path, crate_ty
                 sess.fatal(&format!("failed to write link script to {}: {}", path.display(), e));
             }
 
-            cmd.arg("--script");
-            cmd.arg(path);
+            if sess.target.target.linker_flavor == LinkerFlavor::Gcc {
+                cmd.arg("-Wl,--script");
+                let path_str = if let Some(s) = path.to_str() {
+                    s
+                } else {
+                    sess.fatal(&format!("linker script path {:?} is invalid Unicode", path));
```

Comment:
Does gcc require scripts in Unicode?

Reply:
This is the path, not the contents of the linker script. The issue is that the path is an OsStr, and there's no way to prepend text to an OsStr without converting it to an `&str`.

Reply:
Yes, we can. Convert it to OsString and use push method.

Reply:
I looked at the OsString docs and didn't see that...

Reply:
You didn't see how to convert [OsStr to OsString](https://doc.rust-lang.org/nightly/std/ffi/struct.OsStr.html#method.to_os_string) or the [push](https://doc.rust-lang.org/nightly/std/ffi/struct.OsString.html#method.push) method.

Reply:
Yes, I saw that after you mentioned `push`. I checked the OsString docs while I was writing this code and didn't see it then.

Reply:
Don't worry. Sometimes I do too.